routing/simple.html

```diff
@@ -0,0 +1,240 @@
+        const routes = [response.trip].concat(response.alternates || []);
+        routes.forEach( (route) => {
+          route.legs.forEach( (leg) => {
+            // the route
+            const route_feature = {
+              type: 'Feature',
+              properties: {
+                color: '#' + ('000000' + color.toString(16)).substr(-6),
+                opacity: 0.75,
+                weight: 5
+              },
+              geometry:{
+                type: 'LineString',
+                coordinates: decode(leg.shape, 1e6)
+              }
+            };
+
+            // the origin
+            const origin_feature = {
+              type: 'Feature',
+              properties: {
+                color: '#00bb00',
+                radius: 5,
+                fillOpacity: 0.75
+              },
+              geometry:{
+                type: 'Point',
+                coordinates: route_feature.geometry.coordinates[0]
+              }
+            }
+
+            // the destination
+            const destination_feature = {
+              type: 'Feature',
+              properties: {
+                color: '#bb0000',
+                radius: 5,
+                fillOpacity: 0.75
+              },
+              geometry:{
+                type: 'Point',
+                coordinates: route_feature.geometry.coordinates.slice(-1)[0]
+              }
+            };
+
+            fc.features.push(origin_feature);
+            fc.features.push(destination_feature);
+            fc.features.push(route_feature);
+          }); // legs
+
+          // alternates are lighter
+          color *= Math.floor(color_scale);
+          color_scale = 1;
+
+        }); // routes
+      }); // responses
+
+      //give back the feature collection
+      return fc;
+    }
+
+    function render(req_rep) {
+      const {requests, responses} = req_rep;
+      previous_geojson = geojson;
+      //turn the result into geojson
+      fc = build_geojson(responses)
+      //make a leaflet geojson object
+      geojson = L.geoJson(fc, {
+        style: (feature) => { return feature.properties; },
+        pointToLayer: (feature, ll) => { return new L.CircleMarker(ll, feature.properties); }
+      });
+      //clear the previous
+      if(previous_geojson != null) {
+        previous_geojson.removeFrom(map);
+      }
+      //disappear the input
+      inputs.forEach( (input) => {
+        input.removeFrom(map);
+      });
+      //render the geojson
+      var added = geojson.addTo(map);
+      //fit it in view
+      map.fitBounds(added.getBounds(), { maxZoom: 19 });
+      //update text box and anchor
+      document.getElementById('request_body').value = JSON.stringify(requests);
+      window.history.pushState(null, null, window.location.pathname + '#' + JSON.stringify(requests));
+    }
+
+
+    ///////////////////// EVENTS ////////////////////////
+
+    //build a location object for the request
+    map.on('click', (event) => {
```

Comment:
when you click on the map it creates a marker where you clicked and updates the textbox and the anchor